persistence/persistence-spark/src/main/java/org/cbioportal/persistence/spark/SampleSparkRepository.java

```diff
@@ -0,0 +1,197 @@
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static org.apache.spark.sql.functions.*;
+
+@Component
+@Qualifier("sampleSparkRepository")
+public class SampleSparkRepository implements SampleRepository {
+
+    @Autowired
+    private SparkSession spark;
+
+    @Value("${data.parquet.folder}")
+    private String PARQUET_DIR;
+
+    @Override
+    public List<Sample> getAllSamplesInStudy(String studyId, String projection, Integer pageSize, Integer pageNumber,
+                                             String sortBy, String direction) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta getMetaSamplesInStudy(String studyId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<Sample> getAllSamplesInStudies(List<String> studyIds, String projection, Integer pageSize,
+                                               Integer pageNumber, String sortBy, String direction) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Sample getSampleInStudy(String studyId, String sampleId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<Sample> getAllSamplesOfPatientInStudy(String studyId, String patientId, String projection,
+                                                      Integer pageSize, Integer pageNumber, String sortBy,
+                                                      String direction) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta getMetaSamplesOfPatientInStudy(String studyId, String patientId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<Sample> getAllSamplesOfPatientsInStudy(String studyId, List<String> patientIds, String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<Sample> getSamplesOfPatientsInMultipleStudies(List<String> studyIds, List<String> patientIds,
+                                                              String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<Sample> fetchSamples(List<String> studyIds, List<String> sampleIds, String projection) {
+
+        Dataset<Row> samples = loadStudyFiles(new HashSet<>(studyIds), ParquetConstants.CLINICAL_SAMPLE);
+        
+        if (!CollectionUtils.isEmpty(sampleIds)) {
+            samples = samples.where(samples.col("SAMPLE_ID").isin(sampleIds.toArray()));
+        }
+        
+        samples.createOrReplaceTempView("samples");
+        StringBuilder sb = new StringBuilder("SELECT PATIENT_ID, SAMPLE_ID, studyId");
+        if (PersistenceConstants.SUMMARY_PROJECTION.equalsIgnoreCase(projection)
+            || PersistenceConstants.DETAILED_PROJECTION.equalsIgnoreCase(projection)) {
```

Comment:
This seems odd, in both cases `SAMPLE_TYPE` is appended. Could you cross-verify if this condition is correct.

Reply:
from SampleMapper.xml Line 12 <if test="projection == 'SUMMARY' || projection == 'DETAILED'">. If projection is 'ID' then sampleType won't be included.

Reply:
ok